.github/workflows/reusable-agent-build-container-images.yml

```diff
@@ -75,12 +75,14 @@ jobs:
   # Build cacheable requirements of the images.
   build_image_requirements:
     name: requirements-${{ matrix.builder_name }}-${{ matrix.architecture }}
-    runs-on: ubuntu-22.04
+    runs-on: self-hosted
```

Comment:
For security reasons, we should only use Github cloud and not self hosted runners.